### ISSUE TITLE
Crash: Assertion Failed!

### ISSUE THREAD

**Issue by Adrian-at-CrimsonAuzre** [yegord/snowman#163]
When trying to decompile an exe and it's three dlls, snowman throws this error.

```
Assertion failed!

Program:
C:\snowman-v0.1.2-win-x64\snowman.exe
File:
C:/yegor/buildslave/snowman-master-mingw64/snowman/src/nc/core/igen/IRGenerator.cpp, Line 97

Expression: statement == basicBlock->statements().back()
```

With the prompt to abort, retry, or ignore. Retry forces a crash, and ignore just pulls the message up again.

**Comment by Adrian-at-CrimsonAuzre** [comment]
I simply opened the GUI and gave it an exe that was about 130KB.


-Adrian

**Comment by yegord** [comment]
Could you send this exe file to yegor at derevenets.com?

**Comment by yegord** [comment]
Thanks!
I could not reproduce the issue if I decompile a single exe or dll file.
I managed two reproduce the issue by supplying the three dlls and the exe file at the same time to the decompiler.
Since .text sections of the DLLs occupy overlapping memory regions and the decompiler does not do dynamic linking, most likely these sections blend together and you get nonsense like two or more jumps being at the same address, being put into the same basic block and the basic block has two terminating jump instructions then.
The decompiler probably should not crash but should complain in a more user-friendly way, but you will not get useful results anyway if you do what you do.

**Comment by yegord** [comment]
Yes, these are two overlapping jump instructions ending up in the same basic block:
```
10001094 jnz 0x100010e8
10001095 jnz 0x100010f2
```
